src/Device/Driver/LX_EOS/LXEosDownload.cpp

```diff
@@ -0,0 +1,270 @@
+  data.crc = CalculateCRC(data_array, sizeof(data) - 1, 0xFF);
+  std::span<std::byte> message(data_array, sizeof(data));
+
+  if (!WriteAndWaitForACK(message, env))
+    return false;
+  response[0] = std::byte{ 0x06 };
+
+  try {
+    port.FullRead({ response + 1, sizeof(response) - 1 },
+                  env,
+                  std::chrono::milliseconds(5000));
+  } catch (...) {
+    return false;
+  }
+
+  if (CalculateCRC(response, sizeof(response), 0xFF) != 0x00) // Check CRC
+    return false;
+
+  /* Flight ID for downloading is the index (1 = newest),
+  not the ID from this message */
+  flight.internal.lx_eos.flight_id = index;
+
+  uint32_t julianDate, time_takeoff, time_landing, file_size;
+  std::memcpy(&julianDate, &response[13], 4);
+  std::memcpy(&time_takeoff, &response[17], 4);
+  std::memcpy(&time_landing, &response[21], 4);
+  std::memcpy(&file_size, &response[89], 4);
+  flight.date = julianToDate(julianDate);
+  flight.start_time = BrokenTime::FromSecondOfDay(time_takeoff);
+  flight.end_time = BrokenTime::FromSecondOfDay(time_landing);
+  flight.internal.lx_eos.file_size = file_size;
+  return true;
+}
+
+bool
+LXEosDevice::GetFlightLogBlock(AllocatedArray<std::byte>& block,
+                               uint16_t flight_id,
+                               uint16_t block_id,
+                               OperationEnvironment& env)
+{
+  EosRequestFlightBlock data;
+  data.flight_id = flight_id;
+  data.block_id = block_id;
+  std::byte responseHeader[5];
+  responseHeader[0] = std::byte{ 0x06 };
+
+  auto data_array = reinterpret_cast<std::byte*>(&data);
+  data.crc = CalculateCRC(data_array, sizeof(data) - 1, 0xFF);
+  std::span<std::byte> message(data_array, sizeof(data));
+
+  if (!WriteAndWaitForACK(message, env))
+    return false;
+
+  try {
+    port.FullRead(std::span{ responseHeader + 1, sizeof(responseHeader) - 1 },
+                  env,
+                  std::chrono::milliseconds(5000));
+  } catch (...) {
+    return false;
+  }
+
+  // Read size and ID of received block
+  uint16_t size, id;
+  std::memcpy(&size, &responseHeader[1], 2);
+  std::memcpy(&id, &responseHeader[3], 2);
+
+  // Check that received ID matches the requested one
+  if (id != block_id)
+    return false;
+
+  block.ResizeDiscard(size);
+  try {
+    port.FullRead(
+      { block.data(), block.size() }, env, std::chrono::milliseconds(5000));
+  } catch (...) {
+    return false;
+  }
+
+  std::byte crc_byte[1];
+  try {
+    port.WaitAndRead(crc_byte, env, std::chrono::milliseconds(5000));
+  } catch (...) {
+    return false;
+  }
+
+  // Message has 3 parts: header, block, and CRC byte.
+  // Calculate CRC from all 3 parts, result should be 0x00
+  uint8_t crc = CalculateCRC(responseHeader, sizeof(responseHeader), 0xFF);
+  crc = CalculateCRC(block.data(), block.size(), crc);
+  if (CalculateCRC(crc_byte, 1, crc) != 0)
+    return false;
+
+  return true;
+}
+
+BrokenDate
+LXEosDevice::julianToDate(uint32_t julian_date)
```

Comment:
I think it could be useful to move this to class BrokenDate, e.g. `BrokenDate::FromJulianDay(unsigned)`.